core/feed.go

```diff
@@ -234,21 +252,76 @@ func (t *Textile) feedStackItem(stack feedStack) (*pb.FeedItem, error) {
 }
 
 func (t *Textile) blockIgnored(blockId string) bool {
-	return len(t.datastore.Blocks().List("", -1, "target='ignore-"+blockId+"'")) > 0
+	query := "target='ignore-" + blockId + "'"
+	return len(t.datastore.Blocks().List("", -1, query).Items) > 0
+}
+
+func FeedItemType(item *pb.FeedItem) (pb.Block_BlockType, error) {
```

Comment:
A `FeedItem`'s type is a proto type URL representation of `BlockType`, which can be used (see below) to parse the items when it's type is not known by context. As in, you can switch on all the `BlockType`s.